### PR TITLE
Support URL replacement in errors middleware

### LINKED ISSUE #3420: Custom error pages and original request URL
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I have a frontend written with [Next.js](https://github.com/zeit/next.js/), which contains quite a lot of images and also renders some react stuff on the server. Requests to images can make the Node.js process slower, so I'm thinking of serving them via an nginx container, which is configured to respond to only a folder with images (traefik's `pathPrefix`).

Requests to the folder with images can be combined with a 404 message rendered by Next.js, thanks to the [custom error pages](https://docs.traefik.io/configuration/commons/#custom-error-pages) feature. There is one scenario I cannot get working though:

When nginx returns 404, traefik knocks the Node.js container, but does not pass the original request URL anywhere in the headers. Without this, it is impossible to implement a redirect to a correct image:

```
user
↓ /images/products/PRODUCT_ID-wrong-slug.jpg
traefik
↓
nginx
↓ 404
traefik
↓ /images/products/PRODUCT_ID-wrong-slug.jpg
nodejs
↓ redirect to /images/products/PRODUCT_ID-correct-slug.jpg
traefik
↓
user
```

It'd be great if in addition to `{status}`, it was possible to use `{path}` or `{url}` in `traefik.frontend.errors.XYZ.query`. My nginx container would be configured like this

```
traefik.frontend.errors.all.backend=nextjs
traefik.frontend.errors.all.status=400-599
traefik.frontend.errors.all.query={path}
```
or
```
traefik.frontend.errors.all.query=/custom-error?url={url}
```

(the latter is useful if a website is multilingual and is available at several domain names)

WDYT?

### PR DESCRIPTION
**Pull request by tomMoulard**
### What does this PR do?

This PR adds the `{url}` variable in the `query` option of the ErrorPages middleware.

### Motivation

This PR fixes #3420.

### More

- [X] Added/updated tests
- [X] Added/updated documentation

### CODE REVIEW

File: docs/content/middlewares/http/errorpages.md [hunk 1]
```diff
@@ -121,8 +121,17 @@ The service that will serve the new requested error page.
...
-The URL for the error page (hosted by `service`). You can use the `{status}` variable in the `query` option in order to insert the status code in the URL.
+The URL for the error page (hosted by [`service`](#service))).
+
+There are multiple variables that can be placed in the `query` option to insert values in the URL.
+
+The table below lists all the available variables and their associated values.
+
+|Variable  | Value                                                                         |
+|----------|-------------------------------------------------------------------------------|
+|`{status}`| The response status code                                                      |
```
Comment (kevinpollet) [inline docs/content/middlewares/http/errorpages.md:136]:
```suggestion
|`{status}`| The response status code.                                                      |
```

File: docs/content/middlewares/http/errorpages.md [hunk 2]
```diff
@@ -121,8 +121,17 @@ The service that will serve the new requested error page.
...
+|`{url}`   | The full [QueryEscaped](https://pkg.go.dev/net/url#QueryEscape) requested URL |
```
Comment (kevinpollet) [inline docs/content/middlewares/http/errorpages.md:137]:
```suggestion
|`{url}`   | The full [QueryEscaped](https://pkg.go.dev/net/url#QueryEscape) requested URL. |
```

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by kevinpollet — APPROVED** [review]
Thanks 👍

**Review by mpl — APPROVED** [review]

**Pull request merged**